src/rgw/rgw_gc_log.cc

```diff
@@ -0,0 +1,54 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#include "rgw_gc_log.h"
+
+#include "cls/rgw/cls_rgw_client.h"
+#include "cls/rgw_gc/cls_rgw_gc_client.h"
+#include "cls/version/cls_version_client.h"
+
+
+void gc_log_init2(librados::ObjectWriteOperation& op,
+                  uint64_t max_size, uint64_t max_urgent)
+{
+  obj_version objv; // objv.ver = 0
+  cls_version_check(op, objv, VER_COND_EQ);
+  cls_rgw_gc_queue_init(op, max_size, max_urgent);
+  cls_version_inc(op); // objv.ver -> 1
```

Comment:
@cbodley : While testing deletion of objects inline, I saw that cls_version_inc increases the version to 2 instead of 1. The code first tries to read the version attribute, if it doesnt exist, it sets it to 1 and then increases it, which results in the version being 2 and sets it back. This results in errors in send_chain, which checks for version to be 1 before calling gc_enqueue, hence set_entry is always called. I think this is the reason why I had used set version instead of inc, so I am going to change this part to set version to 1. Is that ok?

Reply:
ah ok, go ahead and change it back!